sim/dex.js

```diff
@@ -752,11 +772,41 @@ class ModdedDex {
 				const bans = banNames.map(v => toId(v));
 
 				if (checkTeam) {
-					ruleTable.complexTeamBans.push([innerRule, '', limit, bans]);
+					let index = -1;
+					for (let i = 0; i < ruleTable.complexTeamBans.length; i++) {
+						if (ruleTable.complexTeamBans[i][0] === innerRule) {
+							index = i;
+							break;
+						}
+					}
+					if (type === '+') {
+						if (index > -1) ruleTable.complexTeamBans.splice(index, 1);
+					} else {
+						if (index > -1) {
+							ruleTable.complexTeamBans[index][2] = limit;
```

Comment:
Are limits not a part of the inner rule? That should probably be fixed. The ideal approach is to treat different limits as entirely different rules.

Reply:
They're not, but different limits shouldn't happen outside of custom formats. Keeping them separate makes it simpler to change a limit.

Reply:
should probably be "remove one limit, add another limit"

Reply:
So you don't have to deal with ambiguity of "which limit overrides which one"